appsmithorg/appsmith
# feat: table widget switch column type

## Description

This PR introduces a new column type: `Switch`. It uses switch widget's component to render the switch inside the table's cell. 

Here is the list of changes that are implemented:
* New column type added: Switch.
* Property pane level options:
     * Computed Value
     * Visible
     * Editable
     * Styles:
          * Horizontal Alignment
          * Vertical Alignment
          * Cell background
     * Events:
          * onChange: This is available only when editable is true
* Table can now be filtered with is Checked and is unChecked conditions

**_NOTE_**: 

* When the Editable property is false then the switches are disabled.
* For the event onChange we are keeping the label as `onChange` in the property-pane but internally we use onCheckChange. Please refer to this [file](https://github.com/appsmithorg/appsmith/blob/b468606a3b354fbe4f55bcae3d56a3faf3ced871/app/client/src/widgets/TableWidgetV2/widget/propertyConfig/PanelConfig/Events.ts#L115).

Please refer to the following for more information:
* [PRD](https://www.notion.so/appsmith/Switch-in-table-5115ccb1a4784530aff55d573f396362)

Fixes #6770 

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

* Cypress spec: `switchCell_spec.js` is added. It covers the following scenarios:
    * Checks if the switch column type is available.
    * Checks if the computed value is editable and `currentRow` object is available.
    * Verifies visibility of the column
    * When editable is `true` then check:
         * If switch is enabled 
         * If `onChange` is available
         * If handler attached to `onChange` is working via interacting with the switch
    * if editable is `false` then verify if switches are disabled
    * Check horizontal, vertical alignment and cell background color
    * Check if the filter conditions: `is checked` and `is unchecked` are working as expected.
    * Verify that `onCheckChange` is not available on `custom columns`

## Checklist:
